vendor/google/gke/node-pool/config/core-pool-sysctl.yaml

```diff
@@ -0,0 +1,37 @@
+kubeletConfig:
+  cpuManagerPolicy: static
+# kubeletExtraConfig:
+#   allowedUnsafeSysctls: 'net.core.*,net.ipv4.*'
+linuxConfig:
+  sysctl:
+   # tune the ipv4 settings to not cause nginx to use all of the tcp memory
+   # addresses: https://jira-secure.berkeley.edu/browse/DH-3
+   #
+   # following this process:
+   # https://cloud.google.com/kubernetes-engine/docs/how-to/node-system-config
+   #
+   # man page:
+   # https://man7.org/linux/man-pages/man7/tcp.7.html
+   #
+   # figures below are measured in units of system page size (4096B), 
+   # and gleaned from the following articles:
+   # https://cromwell-intl.com/open-source/performance-tuning/tcp.html
+   # https://www.ibm.com/docs/en/linux-on-systems?topic=tuning-tcpip-ipv4-settings
+   # https://www.ibm.com/docs/en/linux-on-systems?topic=tuning-network-stack-settings
+   #
+   # net.ipv4.tcp_mem seems to be automagically generated from the supplied tcp_rmem
+   # and tcp_wmem settings.  i believe?
+   #
+   # here be dragons.
+   #
+   # original values (as of 2023-19-04):
+   # net.core.netdev_max_backlog=1000
+   # net.core.rmem_max=212992
+   # net.core.wmem_max=212992
+   # net.ipv4.tcp_rmem=4096 87380 6291456
+   # net.ipv4.tcp_wmem=4096 16384 4194304
+   net.core.netdev_max_backlog: '30000'
+   net.core.rmem_max: '3276800'
+   net.core.wmem_max: '3276800'
+   net.ipv4.tcp_rmem: '4096 87380 16777216'
+   net.ipv4.tcp_wmem: '4096 87380 16777216'
```

Comment:
@shaneknapp I appreciate these comments A LOT right now THANK YOU for writing them alongside this configuration!!

Reply:
thanks @consideRatio !

for some background (that i just realized was omitted from this PR and resides only in our JIRA), this was to help address our nginx ingress controller node running out of tcp kernel memory.  this would, in turn, start serving 500 and 503 to our users, connections would time out and we'd have to kill that pod and things would come back after a restart.

this would happen reliably ~30 days after instruction really kicked off, and due random factor of when, we almost never had a chance to do any forensics on the node.  and it would sometimes happen late in the semester but again, keep the ship afloat, etc etc.

this PR was just a patch to hopefully stop these failures from happening.  i wasn't hopeful but it should *help*...  i think?

in the end, it helped!  about 30 days after heavy usage, all the regular symptoms showed up and users started seeings 503s and timeouts.  about 45 minutes later, the support node was automatically killed and a fresh one cleared things up!  previously this could impacts us for hours, now minutes.

glad you liked the notes in the file.  when you get down in to kerneltuninglandia, there's not a lot of information out there and i knew that if i (or anyone else) needed to poke something this in the future i could at least leave some breadcrumbs.

and to be completely fair:   i still feel that these numbers could be tweaked, but in the end it's still back magic.  :)